### ISSUE TITLE
Support terraform v0.12

### ISSUE THREAD

**Issue by devtook** [terraform-providers/terraform-provider-vsphere#706]
hello,
I use Terraform v0.12.0-alpha4 and Vsphere v1.9.0  to upgrade old code, but tell me Vsphere version is low.
```
terraform.new  0.12upgrade  vsphere/
```
```
Would you like to upgrade the module in vsphere?
  Only 'yes' will be accepted to confirm.

  Enter a value: yes

-----------------------------------------------------------------------------
Error: failed to load provider "vsphere": Incompatible API version with plugin. Plugin version: 4, Client versions: [5]
```

V0.12 will be released, is this a Terragrunt problem or my configuration is wrong?
https://www.hashicorp.com/blog/terraform-0-1-2-preview
https://www.hashicorp.com/blog/terraform-0-12-preview-first-class-expressions
Terraform HCL2 has been greatly improved.

**Comment by lubars** [comment]
Same here - have already converted and tested AWS, GCP, and Azure on 0.12.0-alpha4, now just waiting for vSphere.

**Comment by bill-rich** [comment]
Hi @lubars & @devtook! I created a new branch (`core-v0.12-alpha4`) that you can build from for v0.12.0 support. There is not an official release yet since 0.12 is in alpha, but that will be coming once the release is official.

**Comment by devtook** [comment]
Thanks @bill-rich

**Comment by sofixa** [comment]
@devtook @lubars @bill-rich is the branch work for you? We have a ton of "provider produced invalid plan" errors:

```
Error: Provider produced invalid plan

Provider "vsphere" planned an invalid value for
vsphere_virtual_machine.pa2-front[1].network_interface[1].bandwidth_reservation:
planned value cty.NumberIntVal(0) does not match config value
cty.NullVal(cty.Number).

This is a bug in the provider, which should be reported in the provider's own
issue tracker.


Error: Provider produced invalid plan

Provider "vsphere" planned an invalid value for
vsphere_virtual_machine.pa2-front[1].vapp: attribute representing a list of
nested blocks must be empty to indicate no blocks, not null.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.


Error: Provider produced invalid plan

Provider "vsphere" planned an invalid value for
vsphere_virtual_machine.pa2-front[1].cdrom: attribute representing a list of
nested blocks must be empty to indicate no blocks, not null.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.


Error: Provider produced invalid plan

Provider "vsphere" planned an invalid value for
vsphere_virtual_machine.pa2-front[1].clone[0].timeout: planned value
cty.NumberIntVal(30) does not match config value cty.NullVal(cty.Number).

This is a bug in the provider, which should be reported in the provider's own
issue tracker.


Error: Provider produced invalid plan

Provider "vsphere" planned an invalid value for
vsphere_virtual_machine.pa2-front[1].clone[0].customize: attribute
representing a list of nested blocks must be empty to indicate no blocks, not
null.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.```

**Comment by lubars** [comment]
@bill-rich are your changes included in the new 1.10.0 release?
